src/bootstrap/config.rs

```diff
@@ -537,37 +537,41 @@ impl Config {
         Path::new(&path).components().collect()
     }
 
-    pub fn default_opts() -> Config {
-        let mut config = Config::default();
-        config.llvm_optimize = true;
-        config.ninja_in_file = true;
-        config.llvm_version_check = true;
-        config.backtrace = true;
-        config.rust_optimize = true;
-        config.rust_optimize_tests = true;
-        config.submodules = true;
-        config.fast_submodules = true;
-        config.docs = true;
-        config.rust_rpath = true;
-        config.channel = "dev".to_string();
-        config.codegen_tests = true;
-        config.ignore_git = false;
-        config.rust_dist_src = true;
-        config.rust_codegen_backends = vec![INTERNER.intern_str("llvm")];
-        config.deny_warnings = true;
-        config.missing_tools = false;
-
-        // set by build.rs
-        config.build = TargetSelection::from_user(&env!("BUILD_TRIPLE"));
-        let manifest_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-        // Undo `src/bootstrap`
-        config.src = manifest_dir.parent().unwrap().parent().unwrap().to_owned();
-        config.out = Config::path_from_python("BUILD_DIR");
-
-        config.initial_cargo = PathBuf::from(env!("CARGO"));
-        config.initial_rustc = PathBuf::from(env!("RUSTC"));
-
-        config
+    pub fn default_opts() -> Self {
```

Comment:
I would prefer to not do this, it's not really more readable IMO.

Reply:
reverted